openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainData.java

```diff
@@ -90,17 +105,30 @@ public class SWIGTrainData implements AutoCloseable {
      * After that the BoosterHandle and the DatasetHandle will still need to be initialized at the proper times:
      * @see SWIGTrainData#initSwigDatasetHandle()
      *
+     * TODO: overload this constructor with a default argument fairnessConstrained=false
+     *
      * @param numFeatures   The number of features.
      */
     public SWIGTrainData(final int numFeatures, final long numInstancesChunk) {
+        this(numFeatures, numInstancesChunk, false);
+    }
+
+    public SWIGTrainData(final int numFeatures, final long numInstancesChunk, final boolean fairnessConstrained) {
         this.numInstancesChunk = numInstancesChunk;
         this.swigOutDatasetHandlePtr = lightgbmlib.voidpp_handle();
+        this.fairnessConstrained = fairnessConstrained;
```

Comment:
If we're already using the `ConstraintGroup` moniker on the C++ codebase, where it's easier to mix up with the existing `group` and `constraint` ideas, then let's use the `ConstraintGroup` naming at the provider too, where there are no references to those `group` and `constraint` concepts.